app/views/questions/new.html.erb

```diff
@@ -5,14 +5,35 @@
         <div class="box">
           <h2 class="title-no-margin center">New Question</h2>
           <br>
-            <%= form_for @question do |f| %>
+            <%= form_for(@question) do |f| %>
               <div class="form-group">
-                <%= f.text_field :text, placeholder: 'Question', class: 'form-control', autofocus: true %>
+                <%= f.text_field :text, placeholder: 'Question',  class: 'form-control', autofocus: true %>
               </div>
               <div class="form-group">
-                <%= f.label :qset_id, 'Qset', {class: 'control-label'} %>
-                <%= f.collection_select(:qset_id, Qset.find(current_user.org).descendants, :id, :name, {}, {class: 'form-control'}) %>
+                <button type="button" class="btn btn-primary btn-theme btn-generate-q" data-toggle="modal" data-target="#modal-launch">
+                  Choose Qset
+                </button>
+                <% if !params[:qset].nil? %>
+                  <span class="qset-display">
+                    <%= @qsets.find(params[:qset]).name %>
+                  </span>
+                <% elsif !cookies[:new_question_qset_id].nil? %>
+                  <span class="qset-display">
+                    <%= @qsets.find(cookies[:new_question_qset_id]).name %>
+                  </span>
+                <% else %>
+                  <span class="qset-display no-qset-chosen">No question set chosen</span>
+                <% end %>
+                </span>
+                <!-- hidden input field to pass in qset_id into params on submit -->
+                <% if !params[:qset].nil? %>
```

Comment:
When the qset is preselected by the cookie `#qset-id-input`'s value isn't updated, so the Submit question button is never activated/enabled. We probably need a parallel piece of logic to the `elif !cookies[:new_question_qset_id].nil?` above here.

It might be cleaner/shorter to move the if/elif into the controller, and here (and above, when showing the name in the visible field) use something like 

```
<% if @qset_id %>
  <input type="text" name="question[qset_id]" id="qset-id-input" class="hidden" value="<%= @qset_id %>">
<% else %>
 ...
<% end %>
```

That also allows you to do better error handling -- currently if the user puts in a bad ID, like `9999999` or `apoicjasoidjc`, this causes a 500 error because the template is blindly looking that record up and trying to access the `.name` or `.id` field; if the lookup logic was in the control you could more cleanly trap that error and just unset `@qset`/make it `nil`, so that the regular view logic would effectively ignore bad values.